internal/providers/vault/certificate.go

```diff
@@ -33,24 +38,35 @@ func certificateProvider(ctx context.Context, secretId secrets.SecretIdentifier,
 		return nil, err
 	}
 
-	if secretId.ServerRef == nil {
-		return nil, fnerrors.BadDataError("required server reference is not set")
+	req := certificateRequest{
+		commonName: cfg.GetCommonName(),
+		sans:       cfg.GetSans(),
+	}
+
+	if req.commonName == "" {
+		if secretId.ServerRef == nil {
+			return nil, fnerrors.BadDataError("required server reference is not set")
+		}
+
+		req.commonName = fmt.Sprintf("%s.%s", strings.ReplaceAll(secretId.ServerRef.RelPath, "/", "-"), cfg.GetBaseDomain())
 	}
 
-	commonName := fmt.Sprintf("%s.%s", strings.ReplaceAll(secretId.ServerRef.RelPath, "/", "-"), cfg.GetBaseDomain())
-	return issueCertificate(ctx, vaultClient, cfg.GetIssuer(), commonName)
+	return issueCertificate(ctx, vaultClient, cfg.GetIssuer(), req)
 }
 
-func issueCertificate(ctx context.Context, vaultClient *vaultclient.Client, issuer, cn string) ([]byte, error) {
-	return tasks.Return(ctx, tasks.Action("vault.issue-certificate").Arg("issuer", issuer).Arg("common-name", cn),
+func issueCertificate(ctx context.Context, vaultClient *vaultclient.Client, issuer string, req certificateRequest) ([]byte, error) {
+	return tasks.Return(ctx, tasks.Action("vault.issue-certificate").Arg("issuer", issuer).Arg("common-name", req.commonName),
 		func(ctx context.Context) ([]byte, error) {
 			pkiMount, role, ok := strings.Cut(issuer, "/")
 			if !ok {
 				return nil, fnerrors.BadDataError("invalid issuer format; expected <pki-mount>/<role>")
 			}
 
 			issueResp, err := vaultClient.Secrets.PkiIssueWithRole(ctx, role,
-				schema.PkiIssueWithRoleRequest{CommonName: cn},
+				schema.PkiIssueWithRoleRequest{
+					CommonName: req.commonName,
+					AltNames:   strings.Join(req.sans, ","),
```

Comment:
Weird that they didn't make this a `[]stryng` field…

Reply:
:shrug: